nkms_eth/deployers.py

```diff
@@ -140,45 +213,71 @@ def __init__(self, token_agent):
         super().__init__(blockchain=token_agent.blockchain)
         self.token_agent = token_agent
 
-    def deploy(self) -> Tuple[str, str, str]:
+    def deploy(self) -> Dict[str, str]:
         """
         Deploy and publish the NuCypherKMS Token contract
         to the blockchain network specified in self.blockchain.network.
 
         The contract must be armed before it can be deployed.
         Deployment can only ever be executed exactly once!
 
-        Returns transaction hashes in a tuple: deploy, reward, and initialize.
+        Emits the folowing blockchain network transactions:
+            - MinerEscrow contract deployment
+            - MinerEscrow dispatcher deployment
+            - Transfer reward tokens origin -> MinerEscrow contract
+            - MinerEscrow contract initialization
+
+        Returns transaction hashes in a tuple.
         """
 
-        if self.is_armed is False:
-            raise self.ContractDeploymentError('use .arm() to arm the contract, then .deploy().')
+        # Raise if not all-systems-go
+        self.check_ready_to_deploy(fail=True)
 
-        if self.is_deployed is True:
-            class_name = self.__class__.__name__
-            message = '{} contract already deployed, use .get() to retrieve it.'.format(class_name)
-            raise self.ContractDeploymentError(message)
+        # Build deployment arguments
+        deploy_args = [self.token_agent.contract_address] + self.mining_coefficient    # config
+        origin_args = {'from': self.token_agent.origin}
 
-        deploy_args = [self.token_agent.contract_address] + self.mining_coefficient
-        deploy_tx = {'from': self.token_agent.origin}
+        # 1 - Deploy #
+        the_escrow_contract, deploy_txhash = self.blockchain._chain.provider.\
+            deploy_contract(self._contract_name,
+                            deploy_args=deploy_args,
+                            deploy_transaction=origin_args)
+        deploy_receipt = self.blockchain.wait_for_receipt(deploy_txhash)
```

Comment:
Maybe `_deploy_receipt` for now?  I find myself wondering where it's used (it's not, right?).